pkg/network/tracer/gateway_lookup.go

```diff
@@ -116,8 +116,9 @@ func (g *gatewayLookup) Lookup(cs *network.ConnectionStats) *network.Via {
 			return nil
 		}
 
-		g.subnetCache.Add(r.IfIndex, s)
-		v = s
+		via := &network.Via{Subnet: s}
+		g.subnetCache.Add(r.IfIndex, via)
```

Comment:
This is equivalent now since `Subnet` is the only field in `network.Via`. If we ever add other fields that vary differently from the subnet, then caching `Via` may become a problem.

I'm not necessarily saying we need to change this, but we should be aware of this footgun.